### ISSUE TITLE
Auth0 Social logins do not work with electron 

### ISSUE THREAD

**Issue by Beslinda** [auth0-samples/auth0-electron-samples#12]
I tested the connection to Facebook and Goodle from Auth0 website and it works fine. In my electron app the usual login with email and password works fine but the Social logins do not work. 

In the console I get something like : d.addEventListener is not a function, on s.loginPhonegap . But I don't use phonegap for electron so I am not sure why this does not work

**Comment by chenkie** [comment]
Can you paste the exact output from the console? I was just doing some stuff with Google authentication today and it was working as expected.

![electron-google-signin](https://cloud.githubusercontent.com/assets/1847678/21120537/d51ca034-c095-11e6-8a16-6de2e318f8f7.gif)

**Comment by Beslinda** [comment]
Nothing goes to the console because the error happens as soon as I press login with facebook or google. I am not sure what kind of config I have to add for social login. 

Here it is the lock
	var lock = new Auth0Lock('#id', 'example.eu.auth0.com', {
		auth: {
			redirect: false,
			sso: false
		},
		callbackURL: 'https://example.eu.auth0.com/mobile',
		closable: false,
		displayName: 'example',
	});

**Comment by chenkie** [comment]
Can you show me a screenshot of what you get when you try to log in with social?

**Comment by Beslinda** [comment]
![screen shot 2016-12-13 at 00 46 23](https://cloud.githubusercontent.com/assets/8047976/21121587/a715dd44-c0cd-11e6-944c-59eafff4cb47.png)

**Comment by chenkie** [comment]
Which version of Lock do you have?

**Comment by Beslinda** [comment]
https://cdn.auth0.com/js/lock/10.6/lock.min.js

**Comment by chenkie** [comment]
Ok, try 10.7 and see if there's any difference

**Comment by Beslinda** [comment]
Oh I ok. I will. Thank you

**Comment by Beslinda** [comment]
No I still get the same problem

**Comment by chenkie** [comment]
Ok, it's difficult to tell what it could be. Can you create a repo which reproduces this so I can have a closer look?

**Comment by Beslinda** [comment]
Yes sure I will make a repo later and post it. BTW is there any config for social logins that I have to do or the auth0 takes care of it completely ?

Thank you for your help

**Comment by chenkie** [comment]
The only thing you need to do is add the details for your social connection in the settings area for it in your Auth0 dashboard: https://manage.auth0.com/#/connections/social

With Facebook for example, you need to add your app ID and app secret.

**Comment by Beslinda** [comment]
I have a question which might be the problem. The angular-lock and angular-jwt are node_modules. After the build electron app cannot find those modules (all the modules). If I add the full path. I get the node_modules but other people do not get them.

Do you know what is the problem, why electron app cannot find node_modules after the build ?

**Comment by chenkie** [comment]
It sounds like you may have seen this question that deals with the full path issue: https://stackoverflow.com/questions/30664111/how-to-use-node-modules-within-electron

More detail here: https://stackoverflow.com/questions/32265449/uncaught-error-cannot-find-module-jquery/32335874#32335874

I don't know for sure, but I would imagine you could wire up a build step that would put your required modules into a `dist` folder that can then be shipped with your app. This could be done with webpack for example.

**Comment by Beslinda** [comment]
I tracked down the issue if I remove : 	window.electron = {}; from the script then social login happens but it covers all my electron view and I cannot go back.

Is there a way to fix this ?

**Comment by chenkie** [comment]
Where in sample project is that? I can't see it.

btw, are you using this sample project or are you adapting it to your own? Is there another starter project that you're working from that I can take a look at?

**Comment by tkntobfrk** [comment]
I tried a month or so ago when I noticed custom connections were available in 10.5.x to get an electron + react version working, but never could.  

I didn't have any problems using this git project, it worked without issue. There are issues with electron + Jquery which I haven't solved yet.

@Beslinda It sounds like you are using redirect true?  Have you tried redirect: false.

e.g.
        auth: {
          redirect: false,
          sso: false,
          allowSignUp: false
        }

You could maybe try to eliminate the node integration issue and set  nodeIntegration: false, but I'm not sure if that will help
e.g.:  mainWindow = new BrowserWindow({width: 1050, height: 750,  nodeIntegration: false});

**Comment by Beslinda** [comment]
I have other problems when I set nodeIntegration false. However, I will try to set redirect false and see how it works.

**Comment by chenkie** [comment]
@tkntobfrk what are the issues with jQuery that you've seen?

**Comment by tkntobfrk** [comment]
@chenkie  as a simple example you can setup a twitch 3rd party connection, though I believe I validated this happens with other custom 3rd party connections.

Twitch connection settings: 

user profile script:
function(accessToken, ctx, cb) {
  request.get('https://api.twitch.tv/kraken/user', {
    headers: {
      'Authorization': 'OAuth ' + accessToken,
      'Client-ID': ' ',
      'Accept': 'application/vnd.twitchtv.v3+json'
    }
  }, function(e, r, b) {
    if (e) return cb(e);
    if (r.statusCode !== 200) return cb(new Error('StatusCode: ' + r.statusCode));
    var profile = JSON.parse(b);
    profile.id = profile._id;
    delete profile._id;
    profile.links = profile._links;
    delete profile._links;
    return cb(null, profile);
  });
}

Authorizeurl: https://api.twitch.tv/kraken/oauth2/authorize
Tokenurl: https://api.twitch.tv/kraken/oauth2/token
Scope: user_read channel_read

clone this project
update the auth0 vars in the file, npm start
login successfully the first time. Kill app & npm start again.

click not you
![image](https://cloud.githubusercontent.com/assets/4344301/21289827/51d434f0-c46c-11e6-856d-2e0f63a031a6.png)

You will get forwarded to a page that does nothing
![image](https://cloud.githubusercontent.com/assets/4344301/21289823/1e6bc2a4-c46c-11e6-9105-63c955eeca7a.png)

If you go ahead and set.

  // Create the browser window.
  mainWindow = new BrowserWindow({webPreferences: {
nodeIntegration: false
},width: 800, height: 600});

It will work properly in this case.  There are a bunch of issues with jquery in electron and various solutions on how to properly include jquery so that other windows include the library.  I found none of those solutions to work correctly in some cases.  Disabling nodeIntegration seems to do the trick here.

This is the simplest demonstration, but there are other cases where this can cause problems.

This fork has only the nodeIntegration flag disabled if you want a quick way to test
https://github.com/tkntobfrk/auth0-electron-samples

**Comment by Beslinda** [comment]
In my project I have webviews and when I set nodeIntegration false, the webview does not load.

**Comment by tkntobfrk** [comment]
I had a really funky workaround for this, but I think I have these social logins working based on a fork of https://github.com/iamkevingreen/electron-react-auth0.

Works with Twitch which presented the issue with undefined Jquery https://github.com/tkntobfrk/electron-react-auth0.

Essentially turning nodeIntegration false in main.js and using a preload script to run jquery.

Haven't done much verification, but seems to work.

**Comment by jacwright** [comment]
I've downloaded and installed the example just today and cannot get the facebook/google integrations to work (didn't try twitter). No errors in the console. Has electron or auth0 change something that prevents the happy path from working?

![screen shot 2017-05-09 at 9 05 14 am](https://cloud.githubusercontent.com/assets/120294/25857594/ca82c70a-3496-11e7-89ce-5f8747c6f4ea.png)

**Comment by skunkwerk** [comment]
I am using the sample Electron code as well.
I cannot get Google logins to work.  After I authorize Google, I get redirected to a page like this:

<img width="728" alt="screen shot 2017-08-05 at 1 41 48 pm" src="https://user-images.githubusercontent.com/64326/28998591-f79855ba-79e3-11e7-9325-38db0759ba7f.png">

My settings are:
  var webAuth = new auth0.WebAuth({
    domain: AUTH0_DOMAIN,
    clientID: AUTH0_CLIENT_ID,
    redirect: location.href,
    audience: 'https://' + AUTH0_DOMAIN + '/userinfo',
    responseType: 'token id_token',
    scope: 'openid',
    leeway: 60
  });

And file:/// is in my allowed callbacks.

**Comment by tkntobfrk** [comment]
@skunkwerk - does it work from the console when you test the google auth?  e.g.
https://manage.auth0.com/#/connections/social

**Comment by ArnasAlex** [comment]
When I try an example with username and password - it works. 
Social accounts do not - shows "WE'RE SORRY, SOMETHING WENT WRONG WHEN ATTEMPTING TO LOG IN." without any console error. 
In the Auth0 page under User Details -> History I can see a successful login for that user although it does not work.

**Comment by b4dnewz** [comment]
@skunkwerk I've the same problem as you, I really don't get why... also I recived this answer from a auth0 guy which pointed me to a different example: https://auth0.com/docs/api-auth/grant/authorization-code-pkce but right now..I'm just confused.
